cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -118,27 +122,30 @@ private PushAppResponse writeAppDetailAndPush(RepositoryManager repositoryManage
    * or not.
    *
    * @param repositoryManager the RepositoryManager
-   * @param applicationName the name of this application
+   * @param appRelativePath the relative {@link Path} of the application to write to
    * @return A valid application config file path
    */
-  private Path getAndValidateAppConfigRelativePath(RepositoryManager repositoryManager,
-                                                   String applicationName)
-    throws InvalidPathInSourceControlException, IOException {
-    String fileName = String.format("%s.json", applicationName);
-    Path filePath = repositoryManager.getBasePath().resolve(fileName);
-    if (Files.exists(filePath)) {
-      Path resolvedPath = filePath.toRealPath();
-      Path resolvedRepoRootPath = repositoryManager.getRepositoryRoot().toRealPath();
-      if (!resolvedPath.startsWith(resolvedRepoRootPath)) {
-        throw new InvalidPathInSourceControlException
-          (String.format("Config file %s pointing to invalid path %s", filePath, resolvedPath));
-      }
-      if (!Files.isRegularFile(resolvedPath)) {
-        throw new InvalidPathInSourceControlException
-          (String.format("Config file %s pointing to a directory %s", filePath, resolvedPath));
+  private Path validateAppConfigRelativePath(RepositoryManager repositoryManager,
+                                             Path appRelativePath)
+    throws PushFailureException {
+    Path filePath = repositoryManager.getRepositoryRoot().resolve(appRelativePath);
+    try {
+      if (Files.exists(filePath)) {
+        Path resolvedPath = filePath.toRealPath();
```

Comment:
I thought we had decided not to allow symlinks? Shouldn't this just check that the path is not a symlink and not a directory?

Reply:
Updated the validation to check for symlinks and directory